packages/machine-extractor/src/actions.ts

```diff
@@ -99,6 +99,52 @@ export const ActionAsString = maybeTsAsExpression(
   ),
 );
 
+// List of builtin XState actions that are treated specially in Studio UI
+const SUPPORTED_BUILTIN_ACTIONS = [
+  'xstate.assign',
+  'xstate.log',
+  'xstate.raise',
+  'xstate.stop',
+  'xstate.sendTo',
+];
+
+/**
+ * {type: 'custom name', params: {foo: 'bar'}} Must be extracted as named action
+ * {type: someIdentifier, params: {foo: 'bar'}} Must be extracted as inline action
+ * {type: 'xstate.assign', assignment: {foo: 'bar', baz: () => {}}} Must be extracted as inline action
+ */
+export const ActionAsObjectExpression = createParser({
+  babelMatcher: t.isObjectExpression,
+  parseNode: (node, context): ActionNode => {
+    const id = context.getNodeHash(node);
+    for (const prop of node.properties) {
+      if (t.isObjectProperty(prop)) {
+        if (
+          t.isIdentifier(prop.key) &&
+          prop.key.name === 'type' &&
+          t.isStringLiteral(prop.value) &&
+          !SUPPORTED_BUILTIN_ACTIONS.includes(prop.value.value)
+        ) {
+          return {
```

Comment:
those return values here already can contain `.value` - any reason why `extractEventObject` gets called from `getActionConfig` and not here?

Reply:
`extractObjectRecursively` is called inside `getActionConfig` because we need to access fileContent to slice up values that can't be transferred other than as a string like a regular expression value, etc.